Kibrantn/servicebus/preparer resource caching framework

KieranBrantnerMagee opened the pull request:
Draft PR of enabling automatic resource caching into the preparer framework.

Individual preparers must utilize the set_cache function to define their cache key, the  set of parameters that makes a resource unique.  (This might be SKU, or region, some flags or combinations thereof)

Note: the system currently attempts to be clever and will cache resources that don't have set_cache enabled, but will not be able to ensure the unique parameter alignment within the cache.  (This may be fine for some resources, however, such as resource groups)

YijunXieMS committed:
Fix a sample problem for py2.7 (#9444)

lmazuel committed:
Prepare CI to OpenTelemetry release (#9443)

* Prepare CI to OpenTelemetry release

* Update package_service_mapping.json

mitchdenny committed:
Disable feature CI. (#9416)

Commit:
Manual recovery services (#9446)

* manually generated recovery services

* history and version

* extra line

* Packaging update of azure-mgmt-recoveryservices

* Packaging update of azure-mgmt-recoveryservicesbackup

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

Commit:
fixed history and versoin (#9453)

lmazuel committed:
Fix Readme for OpenTelemetry (#9454)

* Fix Readme for OpenTelemetry

* Update README.md

chlowell committed:
Require TLS for Key Vault challenge authentication (#9457)

KieranBrantnerMagee committed:
Initial commit of preparer resource caching functionality.  Demonstrates use on the servicebus queue tests.

KieranBrantnerMagee committed:
rename test code properly, remove spurious comments.

KieranBrantnerMagee committed:
Adjust test cache levels

praveenkuttappan committed:
Enable import test using devops feed (#8922)

* Enable import test using devops feed

praveenkuttappan committed:
Latest and minimum dependency test (#9136)

* New test to verify against latest released dependent packages

praveenkuttappan committed:
set tox environment based on env (#9283)

chlowell committed:
Revert _BearerTokenCredentialPolicyBase refactoring (#9465)

openapi-sdkautomation[bot] committed:
Generated from ab83411549873547a0a976b1a7e7546fff9e3bad (#9467)

cli generation for subscriptions

changlong-liu committed:
Azurestack (#9473)

* generate azurestack

* add initial HISTORY

changlong-liu committed:
add packaging , ci and dev_requirements (#9475)

changlong-liu committed:
generate initial timeseriesinsights SDK (#9474)

* generate timeseriesinsights

* add initial HISTORY

* add packaging and ci things

* add dev_requirements.txt

Commit:
test generation instruction (#9482)

* test generation instruction

* adding note about service principal

* updated management package release

Commit:
Fileshare refactor to devtools (#9296)

* initial commit

* commit -2

* test directory

* test_share

* test_file and some global refactors

* test_file_service_properties

* test_get_file

* other sync tests

* test_directory_async

* test_file_asyncc

* file_client + file_sp async

* get_file_async

* share_async

* BuildTargetting

* identity

* small change

* more tests

* fix live

* remove teardowns

* more tests

* more changes

* get file async recording

* Fix paralellization

* one more

YijunXieMS committed:
Fix a pylint error (#9486)

openapi-sdkautomation[bot] committed:
Generated from ad0c3f96660205d67ebb9920555493a108b3606e (#9471)

1. Add minSupportedTlsVersion to cluster.json
2. Add example about minSupportedTlsVersion
3. Fixed some resource group name in previous examples

Commit:
history and version (#9488)

* history and version

* Packaging update of azure-mgmt-hdinsight

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

iscai-msft committed:
got rid of unused import (#9484)

YijunXieMS committed:
Enable eventhubs parallel unit tests (#9496)

kristapratico committed:
[textanalytics] adds __repr__ to models (#9442)

* adds reprs to models

* adds test and fixes repr on some models

openapi-sdkautomation[bot] committed:
Generated from 743fe428402054a1d35ba99da95e97a8f2ee7cd5 (#9504)

fix prettier checks on examples & evergrid swagger

Commit:
version and history (#9505)

chlowell committed:
Copy bearer auth base code into Key Vault libraries (#9479)

praveenkuttappan committed:
Enable import test for schedule nightly CI (#9483)

* Enable import test during nightly CI

Commit:
Fix storage tests (#9433)

* retry async

* backoff

* revert

* temporarily skip test

* some tweaks

* one more try

* try except

* increase retry

* fix

* comment

* sync

* oops

* comment

Commit:
network regenerated (#9506)

* network regenerated

* Packaging update of azure-mgmt-network

* regenerated tests

* history and version

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

openapi-sdkautomation[bot] committed:
Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)

updated default tag for app configuration

Commit:
regenerated + change log + version (#9520)

* regenerated + change log + version

* fixed generation mistakes

* regenerated test

* Packaging update of azure-mgmt-cdn

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

changlong-liu committed:
Generate powerbidedicated (#9522)

* genereate sdk

* packaging things

praveenkuttappan committed:
Disable import CI test and run when a variable is set (#9524)

* Disable import CI test and run when a variable s set

scbedd committed:
Single Variable to Identify Test-Mode (#9497)

* adding a default variable to enable the generic preparers to recognize what the livetest mode is

scbedd committed:
Resolve Parallel Windows Failures (#9531)

* non-whl environments should set their install command to have a different pip cache
* create_wheel_and_install must be able to leverage pip cache as well

KieranBrantnerMagee committed:
Merge from master

KieranBrantnerMagee committed:
* Add AreLiveTestsEnabled for use with skipif so that servicebus tests can be selectively disabled and reenabled as they are ported to support Fakes.
* Merge from master
* Adjust dev_requirements for servicebus to support preparers
* Ensure servicebus tests don't use fstrings.

kristapratico committed:
[textanalytics] fix user-agent (#9513)

* fix text analytics user-agent

* fix test

xiangyan99 committed:
Revert "Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)" (#9558)

* Revert "Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)"

This reverts commit 919b13942fad7b68f8bfb9fc10db1f6ce141591f.

* Packaging update of azure-mgmt-appconfiguration

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

rakshith91 committed:
Modify tests.yml for storage (#9516)

* adding live test variables
* setup share
* recording update
* append_block_from_url

scbedd committed:
Run all tests regardless of exit result (#9515)

* all tox instantiations will complete prior to returning the worst return code encountered during the test run.

xiangyan99 committed:
not set "request_id" if it is None (#9547)

* not set "request_id" if it is None

* remove dummy code

* update changelog

* updates

weshaggard committed:
Use common template for SetDevVersion (#9544)

lmazuel committed:
Improve StoragePreparer (#8930)

* Support env variables and stuff

* More tests

* Accept storage account for account_url

* Sync queue shared tests files

* more changes

* Feedback

* Fix queue import

* Fix playback mode

* Updated blob test account URLs

* Updated queue test account URLs

* Secondary endpoints needs init too

* Better support of connection string

* Fixed missing awaits

* Fix file share test pass

* Strip / if endpoints are from ARM GET

Co-authored-by: Rakshith Bhyravabhotla <rakshith.bhyravabhotla@gmail.com>
Co-authored-by: annatisch <antisch@microsoft.com>

mitchdenny committed:
Upgrade doc-warden to 0.5.0. (#9570)

scbedd committed:
moving docsettings out of root, updating analyze.yml to account for new location (#9579)

chlowell committed:
Improve live managed identity tests (#9146)

xiangyan99 committed:
Core enable mypy for customers (#9577)

* Revert "Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)"

This reverts commit 919b13942fad7b68f8bfb9fc10db1f6ce141591f.

* add py.typed

* remove py.typed for tracing

* updates

* include py.typed in wheel package

* fix doc generation

* ignore mypy warning (by design)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#9588)

* Sync eng/common directory with azure-sdk-tools repository

* Clean up docsettings

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

KieranBrantnerMagee committed:
Use existing live_test_only logic instead of novel decorator
thread through random_name_required and force utilization for ResourceGroups via a RandomNameResourceGroup helper in devtools_testutils

praveenkuttappan committed:
Discover and Install prebuilt whl is env variable is set (#9563)

* Discover and Install prebuilt whl is env variable is set

scbedd committed:
Allow Custom Pre-Installs During a Manual Build (#9533)

* adjustments to allow an argument w/ additional packages in it
* adding a wrapper ci.yml that can be invoked against the entire repo
* passing around a BuildDocs param. this means that the test-all can pass 'no docs' but the default will still generate them
* deleting the nightly build ymls. with some clever parametering we don't need them anymore!

KieranBrantnerMagee committed:
Remove deprecated preparer helper imports.

lmazuel committed:
Mock text with the proper API in identity (#9598)

* Mock text with the proper API

* replace lambdas with mock_response helper

Co-authored-by: Charles Lowell <chlowe@microsoft.com>

YijunXieMS committed:
[Event Hubs] migration guide update (#9591)

* migration guide update

* Update anna's feedback

* Correct redesigns to redesign

xiangyan99 committed:
Core proxy doc (#9593)

* Revert "Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)"

This reverts commit 919b13942fad7b68f8bfb9fc10db1f6ce141591f.

* add proxy doc

* update

lmazuel committed:
Fix mock of text in storage (#9606)

scbedd committed:
allow argument at queue time (#9607)

chlowell committed:
Test for azure-keyvault-* 4.0.0 regression in azure-core (#9485)

chlowell committed:
instructions for testing managed identity on VMs (#9354)

praveenkuttappan committed:
Corrected parameter passed to depends test (#9601)

Commit:
generated operations management (#9581)

* generated operations management

* Packaging update of azure-mgmt-operationsmanagement

* add ci.yml

* fixed date in history

* updated readme

* Packaging update of azure-mgmt-operationsmanagement

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

Commit:
regenerated resource (#9582)

* regenerated resource

* Packaging update of azure-mgmt-resource

* history and version

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

kristapratico committed:
[textanalytics] allow rotation of subscription keys (#9406)

* allow rotation and reset of subscription keys

* adds sharedkeycredential

* update samples with sharedkeycredential

* update tests with sharedkeycredential

* make subscription_key _internal

* feedback, added a few tests

* make subscription_key a property

* renames credential class to TextAnalyticsSubscriptionKeyCredential

* update readme

* renames to credential class to TextAnalyticsAPIKeyCredential

chlowell committed:
split deserialization test (#9614)

chlowell committed:
Support matching content in playback (#9587)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#9627)

KieranBrantnerMagee committed:
Merge branch 'kibrantn/servicebus/unify-track2-into-master-final'

kristapratico committed:
[textanalytics] renames TextAnalyticsAPIKeyCredential to TextAnalyticsApiKeyCredential (#9629)

* renames TextAnalyticsAPIKeyCredential to TextAnalyticsApiKeyCredential

* rename samples

praveenkuttappan committed:
Enable import test (#9631)

xiangyan99 committed:
Core type error in deepcopy (#9626)

* Revert "Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)"

This reverts commit 919b13942fad7b68f8bfb9fc10db1f6ce141591f.

* handle TypeError in deepcopy

* update changelog

chlowell committed:
Support client certificate passwords (#9434)

chlowell committed:
Live tests for Cloud Shell managed identity (#9336)

lilyjma committed:
[Cosmos] added readme for samples folder and new sample file (#9537)

* added readme and new sample files

* modified resource id in permission definition

* updated installation instruction in readme

* adjusted omitted_paths to sdk/**/samples/*

Commit:
manually regenerated reservations + history + version (#9632)

* manually regenerated reservations + history + version

* Packaging update of azure-mgmt-reservations

* regenerated after swagger adjustment

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

praveenkuttappan committed:
Update change log when versioning tool is triggered (#9648)

Commit:
fixed failing test (#9647)

bsiegel committed:
Dump deps for graphing (#8616)

danieljurek committed:
update smoke tests to latest version (#9350)

* update smoke tests to latest version

* smoke test storage version 12.0.0 -> 12.1.0

* string formatting that works in python 2.7 along with a couple other potential fixes for a 2.7 KV certs hang on my laptop

* smoke tests: put executing code into a function and call the funciton instead of importing it directly

* install dependencies from dev pipeline

* what version of pip are we running?

* support both pip 19 and 20 when importing pip internals

* fix eventhubs occasional flakyness

* use the track 2 cosmos preview package

kristapratico committed:
[textanalytics] Raise custom error message for AttributeError on DocumentError (#9466)

* adds BatchDocumentErrorException

* reword error and add tests

* more rewording

* add example doc to exception

* fix

* add exception type back in

* change exception back to AttributeError

* feedback and update tests

* fix python 2

* re-run devops with new commit

* raise default AttributeError message myself

* fix tests, need credential class

* update custom error message

YijunXieMS committed:
Improve live test pass rate (#9657)

Commit:
manually generated kusto (#9659)

* kusto historey and version

* Packaging update of azure-mgmt-kusto

* regenerated test

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

Commit:
generated support manually (#9658)

* generated support manually

* updated version

* updated history and version

* space

* udpated packaging

Commit:
Manually generated new version of keyvault (#9641)

* regenerated keyvault

* history and version

* Packaging update of azure-mgmt-keyvault

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

Commit:
fixed location of support (#9681)

* fixed location of support

* Packaging update of azure-mgmt-support

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

Commit:
fixed support packaging (#9682)

* fixed support packaging

* Packaging update of azure-mgmt-support

* just trigger ci again

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

kristapratico committed:
[textanalytics] UX feedback (#9630)

* Renames: NamedEntity/LinkedEntityMatch text->name and LinkedEntity id->data_source_id

* add better type hints to credential parameter

* add service limitations to ref docs

* readme updates

* add better descriptions of models in docstrings

* fix type annotations

* renames: NamedEntity->CategorizedEntity, type->category, subtype->subcategory, data_source_id->entity_id

* renames detect_languages() -> detect_language()

* reanmes document_scores and sentence_scores -> sentiment_scores

* renames LinkedEntity's url -> hyperlink

* removes detected_languages list from DetectLanguageResult

* link out to service limitations

* client docstrings

* adds PiiEntity model

* adds short description of method in README samples

* renames hyperlink->url and entity_id->id (better name pending)

* remove reference to hyperlink in docstrings

* refer to as categorized entities

* Renames SentimentConfidenceScorePerLabel -> SentimentScorePerLabel

* review feedback

YijunXieMS committed:
[Event Hubs] buffer messages to call on_event (sync) instead of passing user callback to uamqp (#9673)

* Buffer message and then call the callback

* test if message buffer is empty

* Retrieve all messages instead of one at a time

praveenkuttappan committed:
Regression test changes (#9640)

* Regression test changes

scbedd committed:
remove pypy3 test run due to imcompatibility with UAMQP (#9680)

chlowell committed:
Simplify and speed up Key Vault test playback (#9608)

chlowell committed:
Mock text with the proper API (#9605)

kristapratico committed:
[textanalytics] remove single text operations (#9639)

* remove single text operations, tests, samples and move some tests to batch

* fix tests

* remove single samples/tests

praveenkuttappan committed:
trigger CI for track2 when tools is changed (#9699)

* trigger CI for track2 when tools is changed

leonardbf committed:
ANF-449 update tests for new SDK (#9460)

YijunXieMS committed:
[Event Hubs] Eventhubs small fixes (#9703)

* Correct dev status

* Fix a doc typo

* Remove task if task is cancelled before consumer is created

* Fix pylint error

Commit:
trying to regenerate resource (#9707)

* trying to regenerate resource

* readded init

* rerecorded tests

* history and version

* Update __init__.py

* Update __init__.py

chlowell committed:
replace is with == (#9540)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#9710)

iscai-msft committed:
[keyvault] added ContentDecodePolicy to pipelines (#9698)

* added ContentDecodePolicy to pipelines for keyvault

* fixed the order the COntentDecodePolicy is added to the pipeline policies

* fixed pylint issues

lmazuel committed:
Test update to support autorest v3 generated client (#9695)

* Allow credential in testframework

* Adapt test framework to support azure-core based Mgmt client

* Fixing more credential things

* Fix missing script declaration

* Update azure_testcase.py

kristapratico committed:
[textanalytics] TextAnalyticsError redesign (#9511)

* simplifies text analytics error and removes inner error model

* adds error target

* adds some tests

* test for invalidDocumentBatch

* add cred class to tests

* adds a few more error cases

* fixes custom AttributeError to use new error design

* fix error tests

kristapratico committed:
[textanalytics] Update history and version (#9696)

* wip history

* updates to history

* more updates to history

praveenkuttappan committed:
Keyvault metapackage is skipped from build (#9714)

* Keyvault meta package should not be skipped from build

praveenkuttappan committed:
Regression test failed to find wheel with dev build version (#9711)

* Regression test failked to find wheel with dev build version

chidozieononiwu committed:
Move Readme and Changelog template from RST -> MD (#9510)

* Move Readme and Changelog template from RST -> MD

* Switch Management HISTORY.rst to CHANGELOG.md

* Include .md in Manifest

* Convert readmes and Update setup.py

* Fix manifest.in and setup.py

praveenkuttappan committed:
Disable pip cache (#9704)

* Create individual pip cache for tox env

openapi-sdkautomation[bot] committed:
Generated from cde744f1992809e6a3df7a483a4e68f0d4d7b381 (#9686)

HanaOnAzure: Update Swagger to correct SKU set

* Remove incorrect SKUs S224oxm, S224oxxm, S224o (they were never released)
* Add SKUs which had not been synced: S224

dw511214992 committed:
update version and changelog (#9722)

* update version and changelog

* Packaging update of azure-mgmt-hanaonazure

* quickfix on format

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

praveenkuttappan committed:
Removed invalid artifact name from ci.yml for azure support service (#9719)

lmazuel committed:
Fix text/plain + BOM in decoder + Response_encoding parameter (#9595)

* Fix #9589

* ChangeLog

* Encoding, finale proposition

* Response encoding

* Feedback

chlowell committed:
Fix AttributeError in async CryptographyClient (#9734)

lmazuel committed:
Storage always returns UTF-8 (#9678)

* Storage always returns UTF-8

* Packaging

* Typo

dw511214992 committed:
Databox sdk release (#9723)

* initial release for databox

* Revert "initial release for databox"

This reverts commit 1e08e5477b7ab8872880541889b16f13ee5cbf6d.

* initial release for databox

lmazuel committed:
Docstring of AzureError (#9737)

* Docstring of AzureError

* Update exceptions.py

anuchandy committed:
Adding placeholder to list any third party licenses (#9746)

praveenkuttappan committed:
Generate dependency graph during analyze step (#9733)

* Generate dependency graph info for track 2 data plane packages

chlowell committed:
Parse App Service MSI responses on Windows (#9480)

chlowell committed:
Manual test for managed identity in Azure Functions (#9689)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#9757)

chidozieononiwu committed:
Convert .rst to .md  (#9548)

* Rst->Md Batch

* RST->Md Core/azure

* RST->Md azure-common

* HISTORY.md -> CHANGELOG.md, azure-core, update setup.py for azure-batch

* HISTORY -> CHANGELOG azure-core-tracing-opencensus

* HISTORY -> CHANGELOG

* rst -> md azure-mgmt

* rst -> md

* HISTORY -> CHANGELOG

* HISTORY -> CHANGELOG

* rst -> md

* add long discription

* Revert azure-mgmt-keyvault to rst

* Revert azure-mgmt-eventhub to .rst

* Ignore readmes in docsettings.yml

* Convert remaining rst to md

Commit:
Fixing multi-api client for keyvault (#9758)

* regenerated again

* history and version

* Packaging update of azure-mgmt-keyvault

* regenerated test

* Packaging update of azure-mgmt-keyvault

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

dw511214992 committed:
Appconfiguration release 2019-11-01 preview (#9762)

* Re-generate integration test
bump version

* Packaging update of azure-mgmt-appconfiguration

* Revert "Packaging update of azure-mgmt-appconfiguration"

This reverts commit 169d6bc698b9fc69b1e3f86fd39107ed0c7550eb.

* Revert "Re-generate integration test"

This reverts commit 0fdbcd17c3525d48fb2c633140dc26757055b513.

* Release for Azure App Configuration - 2019-11-01-preview

* Packaging update of azure-mgmt-appconfiguration

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

dw511214992 committed:
Cognitive release 2017 04 (#9764)


* Release management SDK for cognitive services

* Packaging update of azure-mgmt-cognitiveservices

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

dw511214992 committed:
Release for ADF - 2018 - 06 (#9769)

* Release for ADF - 2018 - 06

* Packaging update of azure-mgmt-datafactory

* change version

* add breading changes note to CHANGELOG.md

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

chlowell committed:
Distinguish credential unavailability from failure (#9372)

chlowell committed:
Adjust tests for soft-delete enabled by default (#9770)

chlowell committed:
clear os.environ when patching it (#9771)

chlowell committed:
Prevent raising KeyVaultErrorException (#9716)

xiangyan99 committed:
Core doc (#9752)

* Revert "Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)"

This reverts commit 919b13942fad7b68f8bfb9fc10db1f6ce141591f.

* core doc update

praveenkuttappan committed:
Edit setup py to install dev version of required package (#9768)

* Edit setup py to install dev version of required package and ensure required packages are available before running depends test

kristapratico committed:
[textanalytics] Fix live test framework (#9777)

* fixes to test framework b/c new release of azure-mgmt-cognitiveservices

* fix for recorded tests and service bug failing live tests

* actually fix live/recorded tests

* also fix for legacy tests

kristapratico committed:
update readme with logging info (#9731)

00Kai0 committed:
Test databoxedge (#9767)

* add base test.

* add test for databoxedge

00Kai0 committed:
minimal test for databox (#9785)

kristapratico committed:
fix inner error check (#9787)

xiangyan99 committed:
Core release date feb (#9793)

* Revert "Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)"

This reverts commit 919b13942fad7b68f8bfb9fc10db1f6ce141591f.

* update release date

chlowell committed:
azure-keyvault-* 4.0.1 (#9794)

chlowell committed:
Update azure-identity for release 1.3.0 (#9797)

xiafu-msft committed:
[ADLS Gen2]Add ADLS Async (#9541)

chlowell committed:
Update Key Vault 4.0.1 release date (#9802)

chlowell committed:
Live tests for AKS managed identity (#9495)

xiafu-msft committed:
[Storage][ADLS][History]Add history (#9805)

heaths committed:
Add security policy (#9806)

kristapratico committed:
ux debrief feedback (#9804)

changlong-liu committed:
Manually gen v7.2.0 for azure-mgmt-storage (#9789)

00Kai0 committed:
add minimal test for servicefabric (#9810)

kristapratico committed:
ignore repr on the client (#9816)

kristapratico committed:
typo in readme (#9817)

praveenkuttappan committed:
Fixed path issue when searching for change log path (#9813)

praveenkuttappan committed:
Regression test should not fail if required package is not available (#9778)

* Exit gracefully when required package is not available on pypi
* Create local branch from fetch_head

openapi-sdkautomation[bot] committed:
Generated from 7a434dc02be9f218af645a7e41e602403658e4e2 (#9827)

put the latest version to the top

dw511214992 committed:
update changelog and version of databoax (#9828)

xiangyan99 committed:
Identity app service msi tests (#9727)

* app service msi tests

annatisch committed:
[Storage] stg71 (#9611)

* [Storage][FileShare]regenerate swagger code (#9022)

* Readme transform (#9107)

regenerate code

* [Storage][FileShare]Regenerate code 😎 (#9231)

* File handles (#9041)

* [Storage][FileShare][STG71]Close File Handles

* [Storage][FileShare][STG71]Add attr for list shares/get share properties (#9071)

* [Storage][FileShare][STG71]Add attr for list shares/get share properties

* [Storage][Blob]Regenerate Code (#9233)

* [Storage][FileShare][STG71]copy file acl and attributes (#9142)

* [Storage][FileShare][STG71]copy file acl and attributes

* fix error

* fix pylint

* [Storage][Blob][STG71]CPK-N (#9202)

* [Storage][Blob][STG71]CPK-N

* [Storage][FileShare][BugFix]Clear Page is not working 😱 (#9108)

* generate code

* [Storage][File]Clear Range Test

* [Storage][FileShare][STG71]file lease (#9132)

* [Storage][Blob][STG71]Managed Disk Page Range Diff 💿 (#9264)

* [STG71]Edit HISTORY.md and _version (#9317)

* [STG71]Edit History and Version

* [Storage] Migrate file-share tests (#9576)

* Merged blobs and queues

* Merge unchanged file tests

* Merged share tests

* Merged handle tests

* Removed filetestcase

* Merged sync file tests

* Merged async file tests

* Some test cleanup

* Resolve skipped file tests

* [FileShare]un-skip and re-record the tests (#9619)

* [Storage] API version validation (#9635)

* Expose api version parameter

* Basic API version check

* 2nd iteration with decorators

* Removed API version setter

* Synced shared folder

* Updated blob API

* Updated blob tests

* Pylint

* Added API version to file shares

* Support API version in queues

* File client API validation

* Reverted API change for now

* Removed version check

* Fix API check tests

* Pylint

* [STG71][FileShare]rename ShareFileLeaseClient to ShareLeaseClient (#9728)

* [FileShare][Lease]add Lease info to FileProperties (#9759)

* [Storage][Doc]Add Logging to README.md (#9780)

* [Storage] Storage stg71 FileShares API review updates (#9772)

* Updated CopyFileSmbInfo behaviour

* Missing parameter docs

* Update tests

* Pylint

* Fix file permission default

* Flatten SMB parameters

* Renamed permission key

* Update changelog

* [Storage][Blob]mark managed_disk_diff test playback only (#9820)

It's complicated to set up a managed disk account

* [Storage] Storage stg71 Blobs API review updates (#9691)

* Flatten scope models

* Fix test

* Turned managed disk diff into parameter

* Removed managed disk support

* Updated container encryption

* Fix pylint

* Update behaviour of snapshot url

* ContainerEncryptionScope rename

* Update container cpk tests

* Pylint

* Update managed disk api

* Updated readme

* Update docs

* [Storage] Documentation updates (#9825)

* Updated blobs docs API version

* Updates to blob docs

* Fix some docstrings

* Added more file share docs

* Last docs updates

* API version check

* Missing version tags

Co-authored-by: Xiaoxi Fu <49707495+xiafu-msft@users.noreply.github.com>

chlowell committed:
Adopt vcrpy 3.0.0 (#9747)

xiafu-msft committed:
[FileShare]re-record (#9835)

danieljurek committed:
also install pkginfo (#9839)

xiafu-msft committed:
[FileShare]re-record (#9842)

praveenkuttappan committed:
Manual version bump for azure-core and azure-ai.textanalytics (#9822)

iscai-msft committed:
added kwargs to ContentDecodePolicy in azure core (#9845)

iscai-msft committed:
Add RequestIdPolicy to default policies (#9841)

praveenkuttappan committed:
Install azure dev tools and sdk tools as part of CI tools (#9836)

* Instal azure dev tools and sdk tools as part of CI tools
* Fix nested dependency issue by skipping already installed packages present in dev requirement

00Kai0 committed:
minimal test for storagesync (#9829)

* add minimal test for storagesync

* add test about syncgroup in storageasync

chlowell committed:
Consolidate azure-keyvault-keys shared test code (#9800)

xiafu-msft committed:
Update CHANGELOG.md (#9860)

lmazuel committed:
Delete is not LRO (#8869)

praveenkuttappan committed:
Fix to install dev tools from test repo inside virtual env (#9863)

* Fix to install dev tools from test repo inside virtuall env

changlong-liu committed:
Sdk 0.13.0 for azure mgmt hanaonazure (#9852)

* add version number 0.13.0

* gen sdk

praveenkuttappan committed:
Upload dependency graph and add badge (#9799)

* Upload dependency graph and add badge

praveenkuttappan committed:
Manual version and change log udpate after release (#9847)

YijunXieMS committed:
[Event Hubs] Increment version after release (#9870)

* Increment version

* Update Development Status

* Update bug fix section

yunhaoling committed:
[ServiceBus] Update source url and version number in readme (#9873)

changlong-liu committed:
Manually release Containerservices v8.3.0 (#9875)

* generate containerservice sdk

* release version 8.3.0

HankiGreed committed:
Simplify ManagedIdentityCredential

praveenkuttappan committed:
Pin idna version to 2.8 until new version of requests package is released (#9888)

chlowell committed:
Consolidate azure-keyvault-secrets shared test code  (#9859)

chlowell committed:
Consolidate azure-keyvault-certificates shared test code  (#9861)

chlowell committed:
update CryptographyClient docstrings (#9891)

JimSuplizio committed:
Plumb ServiceDirectory parameter variable through release pipelines (#9894)

lmazuel committed:
Exception 2.0 prototype (#9738)

* Exception 2.0 prototype

* Mypy

* Accept "model" kwargs

* added kwargs to map_error to pass model

* Revert "added kwargs to map_error to pass model"

This reverts commit 2cf06c5b81bcc67ec852c8136e8da8f8a96bf20b.

* Exceptions update

* More tests on deprecation

* Blackened

* Be a little less resilient on OData format, to avoid empty messages

* Make ODataV4Format accessible

* Pylint

* Save some space

Co-authored-by: iscai-msft <43154838+iscai-msft@users.noreply.github.com>

YijunXieMS committed:
[Event Hubs] Fix bug in closing connection upon exceptions (#9617)

* Fix bug in closing connection upon exceptions

* close_handler for EventHubError and merge exception handling clause

* Fix a pylint error

weshaggard committed:
dependencies -> dependency

00Kai0 committed:
add test for cdn, coverage: 33/41. (#9886)

kaerm committed:
Update root README (#9826)

- Making it more accessible to people consuming this page on azure.github.io to avoid redirecting to github from docs
- Updating the description to point people to the most recent packages

KieranBrantnerMagee committed:
Transition examples folder to samples folder and other assorted SB cleanup (#9679)

* Samples and structural cleanup: Move examples folder to samples folder, adjust docstrings, conftest, yml, manifest, and imports.
* Brings over a more fully featured get_running_loop from uamqp to address odd behavior in some historical versions
* Populate readme with some basic info taken wholesale from EventHubs, remaining samples need to be documented and linked in.  
* Additionally took EH's structure of sync_samples in their own folder as well.
* Add .. admonition:: in front of Example: in docstrings
* Adjust examples -> samples in setup.py exclude stanza

kristapratico committed:
[textanalytics] allow user to specify "none" for country hint (#9889)

* allow user to specify none for country hint

* update docs

* fix and add async test

praveenkuttappan committed:
Graceful exit when version increment is not supported (#9655)

kristapratico committed:
[textanalytics] More naming changes (#9929)

* rename SentimentScorePerLabel to SentimentConfidenceScorePerLabel

* rename sentiment_scores to confidence_scores

* rename LinkedEntity.id to LinkedEntity.data_source_entity_id

* update history and feedback

lmazuel committed:
Synapse dataplane (#9892)

* Initial generation Synapse autorest v5

* Fix empty model generation

* Fix default value for parametrized host

* Update Synapse DNS

* Samples

* Doc fixes

* Release date

* Github owner

* Shared req

* Packaging

* Synapse DataPlane EngSys (#9932)

* necessary engsys components to release

* allow pylint failure. add readme exception

* update manifest to include samples in sdist

* last analyze failure hopefully

Co-authored-by: Laurent Mazuel <lmazuel@microsoft.com>

* Revert "Shared req"

This reverts commit ae45761d5c0c27efe68ce06142d667a998b15a91.

Co-authored-by: Scott Beddall <45376673+scbedd@users.noreply.github.com>

praveenkuttappan committed:
Fix random failures on CI due to wheel and sdist build for dev required packages (#9942)

chlowell committed:
Verify JSON attributes in Key Vault test playback (#9918)

chlowell committed:
Add context manager API to Key Vault clients (#9906)

rakshith91 committed:
Blobs - Fix Passing Tranport in get_blob_client (#9954)

kristapratico committed:
update text analytics codeowner (#9956)

praveenkuttappan committed:
Install sdist from pre-built package (#9951)

* Install sdist from prebuilt package

scbedd committed:
new released opentelemetry api removes an api we use (#9962)

kristapratico committed:
[textanalytics] Feedback changes (#9955)

* change cred check to hasattr api_key

* make country_hint and language keyword arguments

* add some tests to check for keywords

* fix pylint

LianwMS committed:
Manually release TimeSeriesInsights 0.2.0 (#9883)

Merge

KieranBrantnerMagee committed:
- Prevent inferred caching. (You now have to explicitly declare a parent resource is cached if a child is cached, otherwise an exception will be thrown before test run)
- Aggregate cache key with parents, such that, for instance, a service bus underneath a EU RG won't be swapped for a service bus underneath an NA RG. (Previously only took into account child cache settings to look up the child in the cache)
- Switch cache usage in test classes to use decorators for ease, add helper to resource group preparer.
- Clean up logging and documentation, tweaks to ensure pylint is happy.
- Add azure-devtools to dev-requirements
- Remove spurious delete cleanup code.

YijunXieMS committed:
[Event Hubs] Add security token to management request message application properties (#9931)

* Increment version

* Update Development Status

* Add security_token to management request message's application_properties

* Fix an async call issue

* use auth.token instead of get_token()

chidozieononiwu committed:
Update MacOS VM Image (#9933)

KieranBrantnerMagee commented on tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py:
```diff
@@ -59,18 +112,47 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
+                elif test_class_instance._delay_removal: # pylint: disable=protected-access
+                    # If a child is specifying a cache but we aren't, raise an exception.
+                    # If it were ever desired to do implied caching, one would simply add a deferred delete here.
+                    # raise Exception("Preparer Error: If a child resource is cached, parents must also be cached")
```
Note to self:  This comment can be removed and replaced with a comment denoting that this section is relevant if the check on line 58 ever goes away.

KieranBrantnerMagee commented on tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py:
```diff
@@ -36,20 +50,59 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
+            child_is_cached = getattr(fn, '__use_cache')
+            # Note: If it is ever desired to make caching inferred, remove this if/throw.
+            # This ensures that a user must _very specifically say they want caching_ on an item and all parents.
+            if not self._use_cache:
```
Note to self:  add "child_is_cached" to this if; even though it's implied in getting to this point since getattr would fail, being explicit would improve code clarity and robustness.

praveenkuttappan committed:
Skip deleting build and egg info directories in CI (#9980)

* Skip deleting build and egg info directories in CI

scbedd committed:
Pinning opentelemetry sdk and api in opentelemetry (#9978)

* bumping preview version of the package. locking opentelemetry-api to version 0.3a0 in both setup.py and dev_requirements

xiangyan99 committed:
add request hook support (#9958)

* add request hook support

* update tests and changelog

LianwMS committed:
Manually release for rdbms (#9944)

* Generated from 2e95c3fc5d0af2b710b847bb737a7848505e17cc

Removing quotes

* Manually release for PostgreSQL/MySQL/MariaDB

Removing quotes

kristapratico committed:
[textanalytics] create reference links in README (#9988)

* create reference links

* use relative links where possible

lmazuel committed:
Remove explicit version tag from automation (#10002)

* Remove explicit version tag from automation

* Update swagger_to_sdk_config.json

00Kai0 committed:
add test for msi (#9997)

00Kai0 committed:
add test for databricks (#9998)

rakshith91 committed:
Storage - use user agent policy from azure-core (#9985)

* User Agent policy for blobs

* more changes

* lint + tests fix

* remove policy

* update changelog

* small change

* comments

* update base client

* lint

* more lint

* undoe recordings

* update change log

danieljurek committed:
add resource group provisioniner and deprovisioner (#9798)

* add resource group provisioniner and deprovisioner
* with tags
* use condition for allocating resource group, configure storage to skip allocationg an RG, collapse KV configuration now that all can run in parallel
* keep KV tests parallelized per-package
* remove extra parens
* correct template expression syntax
* correct syntax for second conditional expression
* set resource group allocation for live tests for services whose live test migration status is known
* servicebus: eliminate old credentials
* Event Hubs is using a static resource today
* add identity configuration
* always remove the resource group when it has been created successfully
* remove AZURE_CLIENT_OID from service bus tests.yml

scbedd committed:
update core test to properly execute on py2.7 (#10007)

Commit:
generated synapse (#9999)

* generated synapse

* Packaging update of azure-mgmt-synapse

* updated pipeline and changelog

* + minor updates

* Packaging update of azure-mgmt-synapse

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

scbedd committed:
make certain the job will fail if there are test failures recorded (#10016)

annatisch committed:
[Storage][Queue] Fix encode policy docs (#10031)

* Fix encode policy docs

* Fix async encode policy docs

kristapratico committed:
[textanalytics] Response hook inherits from SansIOHTTPPolicy (#9984)

* fixing response hook policies (wip)

* fix for async response hook

* rename to raw_response_hook to be consistent with azure-core

* fix sample

chlowell committed:
Prevent unecessary Key Vault poller sleeping (#9991)

praveenkuttappan committed:
Skip build and egg info folders from pytest collection lookup (#10048)

* Skip build and egg info folders from pytest collection lookup
* add ignore arg to sdist

rakshith91 committed:
Fix Datalake tests (#10047)

* Fix Datalake tests

* update recording

* one more change

Commit:
regenerate version of cosmosdb (#10032)

* regenerate version of cosmosdb

* fixed generation mistake

* Packaging update of azure-mgmt-cosmosdb

* regenerated test

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#9915)

xiafu-msft committed:
[ADLS]Add Set/Get File System ACL (#10017)

xiangyan99 committed:
add policy doc (#10001)

* add policy doc

* capitalize policies

kristapratico committed:
rename length/offset to grapheme_length and grapheme_offset (#9993)

Commit:
regenerated storage (#10034)

* regenerated storage

* Packaging update of azure-mgmt-storage

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

Commit:
regenerated compute (#10033)

* regenerated compute

* Packaging update of azure-mgmt-compute

* missing apis

* updated changelog

* regenerated tests

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

lmazuel commented on tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py:
```diff
@@ -5,17 +5,28 @@
 class AbstractPreparer(object):
+    _cache_lock = Lock()
+    _resource_cache = {}
+    _cache_pending_deletes = {}
```
Being you clean using "values()", order won't be respected. I would more imagine a stack for LIFO behavior

KieranBrantnerMagee replied:
a queue complicates the process of deletion, largely because I was relying on the keying as a crutch to prevent duplicate deletes, although I could _probably_ be a bit more precise with only adding to the deletion cache on first creation which would maintain the "only once" nature.   Will try to clean this up++

(And if I want to be tongue-in-cheek about this, with python 3.7+ we _can_ treat dict values() as LIFO so long as I reverse before deleting :P)

lmazuel commented on tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py:
```diff
@@ -36,20 +50,59 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
-                except AzureNameError:
-                    if i == retries - 1:
+        # If a child is cached we must use the same cached resource their equivalent parent did so all the deps line up
+        try:
```
I would prefer to see this block in "_preparer_wrapper" and not add "fn" as a parameter of "_prepare_create_resource". I actually split "_preparer_wrapper" into "_preparer_wrapper/_prepare_create_resource" not that long ago because I wanted to have some preparation code available without having to fit "fn"

lmazuel commented on tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py:
```diff
@@ -36,20 +50,59 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
+            pass
+
+        # We must use a cache_key that includes our parents, so that we get a cached stack
+        # matching the desired resource stack. (e.g. if parent resource has specific settings)
```
smart :)

lmazuel commented on tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py:
```diff
@@ -59,18 +112,47 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
+                    # latter cache-fetched-instances didn't actually initialize ARM clients to be able to delete.
+                    if self._aggregate_cache_key not in AbstractPreparer._cache_pending_deletes:
+                        with self._cache_lock:
+                            AbstractPreparer._cache_pending_deletes[self._aggregate_cache_key] = _deferred_delete
```
Why do you need "_cache_pending_deletes" and not just walk through "_resource_cache" ?
And why a "_deferred_delete" with closure, and not directly call "remove_resource" on the cached resource? We could cache the first preparer, so we would have enough context to clean it directly. Thought?

KieranBrantnerMagee replied:
Initially that was what I tried to do, actually.  However, I ran into some _very strange_ issues with regards to some properties on that entity (e.g. namespace client) being None, that didn't occur when I used the closure instead.  I'm inclined to think in hindsight that I was in some way doing what you imply (e.g. not actually storing the Proper Original preparer), so I can revisit that now.  If it walks into the same failure mode again though I may reassess to avoid ratholing.

lmazuel commented on tools/azure-sdk-tools/devtools_testutils/resource_testcase.py:
```diff
@@ -41,6 +43,7 @@ def __init__(self, name_prefix='',
             self._need_creation = False
         if self.random_name_enabled:
             self.resource_moniker = self.name_prefix + "rgname"
+        self.set_cache(use_cache, location)
```
"location" is actually fake in ARM, a resource group is a concept and doesn't exist physically..

I think what you want is to be sure you create two resource groups in this scenario:
```python
@ResourceGroupPreparer(use_cache=True)
@ResourceGroupPreparer(use_cache=True, parameter_name="rg2")
```

which is here would use the same cache.

The only reason someone would want to change the "parameter_name" would be for multiple RG (since this impacts the name of the variables you inject in the tests, this is actually the ONLY way to have two RGs)

I think it means to me the default cache_key should use the parameter_name too

KieranBrantnerMagee replied:
++ Good callout both on location and on parameter_name being the actual differentiator, you hit the nail on the head in both cases WRT my intent.

lmazuel commented on tools/azure-sdk-tools/devtools_testutils/resource_testcase.py:
```diff
@@ -81,3 +84,5 @@ def remove_resource(self, name, **kwargs):
             except CloudError:
                 pass
+
+CachedResourceGroupPreparer = functools.partial(ResourceGroupPreparer, use_cache=True)
```
Let's activate the "random_enabled" as well?

lmazuel commented on tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py:
```diff
@@ -59,18 +112,47 @@ def _prepare_create_resource(self, test_class_instance, **kwargs):
             trim_kwargs_from_test_function(fn, kwargs)
 
             try:
+                test_class_instance._delay_removal = False # pylint: disable=protected-access
```
Could you tell me how you use that?

KieranBrantnerMagee replied:
So this may not be relevant given your constraint of requiring parent elements to declare their cache status;  I'm inclined to keep it in however similar to my comment above of "if we ever go to that modality this suddenly becomes important again"

In that case it's used to "inform" parent classes that otherwise wouldn't have known they should be cached, that they shouldn't delete themselves until the end lest they nuke the cached sub-resources in that process.

(I'm realizing via typing this that it's probably duplicating functionality between the __use_cache tag on fn, so I'll replace this with that accordingly)

lmazuel committed:
Fix azure-core dep with get_client_from_cli_profile (#10043)

* Fix azure-core dep with get_client_from_cli_profile

* Packaging

* Remove AccessToken from the doc

* Add basic credentials test

* More tests

* Typo

* azure-common needs azure-core for testing

* Today

scbedd committed:
Update pathlib requirement appropriately (#10073)

* pinning a ci_tools requirement for importlib-resources

Commit:
sql - manually regenerated (#10068)

* manually regenerated

* Packaging update of azure-mgmt-sql

* updated changelog

* fixed sql

* regenerated test

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

Commit:
container registry regenerated (#10067)

* container registry regenerated

* Packaging update of azure-mgmt-containerregistry

* test

* reformatted

Co-authored-by: Azure SDK Bot <adxpysdk@microsoft.com>

xiangyan99 committed:
Core stablize tests (#9890)

* Revert "Generated from 1f5a1bea9d656db85e7285f259aced9f48278b45 (#9519)"

This reverts commit 919b13942fad7b68f8bfb9fc10db1f6ce141591f.

* stablize tests

praveenkuttappan committed:
Update to latest doc-warden and change package requirement to pathlib2 (#10081)

* Update to latest doc-warden and change package requirement to pathlib2

praveenkuttappan committed:
Unpin idna to use latest version (#10062)

* Unpin idna to use latest version

chlowell committed:
Update Key Vault changelogs and versions (#10080)

xiafu-msft committed:
[ADLS][BugFix]Datalake Doesn't support Mooncake (#10089)

iscai-msft committed:
[TEXT ANALYTICS] Split up tests into endpoints, increase test coverage (#10057)

* split up sync tests

* split up async tests

* rerecorded tests

* implemented Krista's comments

* integrated grapheme changes into split up test infrastructure

* added back test_out_of_order_ids

* corrected function calls

* added newline to test files

* recordings for test_auth

* removed test_length_with_emoji

* removed live test only auth recordings

* ensure there is a correct response when some of the passed in inputs are errors

* renamed some test names

* re-recorded tests

Commit:
Prevent flaky test from running on Emulator (#9919)

* fixed flaky test

* added cleanup script

* moved cleanup.py to a higher folder level

* removed configs

* set database id from environment varaiable

* remove init file from tests

* add in prep and cleanup steps

* fully parallel matrix, use az cli to delete database

* add always() condition to clean up when build is canceled

* follow deprecation guidance; use 'cosmos sql database'

Co-authored-by: Daniel Jurek <djurek@microsoft.com>

xiafu-msft committed:
[ADLS]Add get_root_directory_client API (#9938)

xiafu-msft committed:
[Queue][BugFix]send/message Response not Accurate (#10101)

Commit:
datalake tests fix (#10104)

xiangyan99 committed:
Add max_timeout support in retry_policy (#10011)

* Add max_timeout support in retry_policy

* update changelog

KieranBrantnerMagee committed:
Create a new helper in azure_testcase, get_replayable_random_resource_name, to be used by mgmt tests to properly fetch names in live vs nonlive scenarios while still aligning recordings. (#10082)

Emplace this helper in servicebus management tests and regenerate recordings.
Ensure servicebus preparers use the scrubber to properly align live and nonlive test names.

KieranBrantnerMagee committed:
Isolate cache code out of prepare_create_resource
remove the need for a distinct cache_pending_deletes structure
update servicebus dev requirements
add random_name_enabled to resource_testcase cached helper wrapper.
fix bug in how trimmed_kwargs are used such that we do not trim the kwargs that we pass to deletion, since we may remove elements needed for removal.

lmazuel committed:
Fix azure-core odata weird scenarios (#10116)

* Make ODataV4Format more robust to internal errors

* ChangeLog

KieranBrantnerMagee committed:
Merge from upstream master

KieranBrantnerMagee committed:
Move resource cache cleanup into base conftest

KieranBrantnerMagee commented:
Given that the initial unification branch this was merging into dealt with long before this got in, I've created the formal PR in the main repository here: https://github.com/Azure/azure-sdk-for-python/pull/10126

(Thanks again @lmazuel  for all your help on this and the initial sanity check)
